docs/site/Interceptors.md

```diff
@@ -0,0 +1,328 @@
+    return `Hello, ${name}`;
+  }
+
+  @intercept(logSync)
+  greetSync(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  greet(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept('log')
+  async hello(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  async greetWithDI(@inject('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept('log', logSync)
+  async helloWithTwoInterceptors(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  async helloWithoutInterceptors(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(changeName)
+  async helloWithChangeName(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(logWithErrorHandling)
+  async helloWithError(name: string) {
+    throw new Error('error: ' + name);
+  }
+}
+```
+
+### Class level interceptors
+
+To apply interceptors to be invoked for all methods on a class, we can use
+`@intercept` to decorate the class. When a method is invoked, class level
+interceptors (if not explicitly listed at method level) are invoked before
+method level ones.
+
+```ts
+@intercept(log)
+class MyControllerWithClassLevelInterceptors {
+  static async greetStatic(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  static async greetStaticWithDI(@inject('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  @intercept(logSync)
+  greetSync(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(convertName, log)
+  async greet(name: string) {
+    return `Hello, ${name}`;
+  }
+}
+```
+
+### Global interceptors
+
+Global interceptors are discovered from the `InvocationContext`. They are
+registered as bindings with `interceptor` tag. For example,
+
+```ts
+app
+  .bind('interceptors.MetricsInterceptor')
+  .toProvider(MetricsInterceptorProvider)
+  .tag('interceptor');
+```
+
+### Order of invocation for interceptors
+
+Multiple `@intercept` decorators can be applied to a class or a method. The
+order of invocation is determined by how `@intercept` is specified. The list of
+interceptors is created from top to bottom and from left to right. Duplicate
+entries are removed from their first occurrences.
+
+Here is the list of interceptors invoked for each method on
+`MyControllerWithClassLevelInterceptors`.
+
+- greetStatic: `log` (from the class)
+- greetStaticWithDI: `log` (from the method, class level `log` is removed)
+- greetSync: `log`, `logSync` (from the method, method level `log` is preserved
+  and class level `log` is removed)
+- greet: `convertName`, `log` (from the method, method level `log` is preserved
+  and class level `log` is removed. This allows a method to explicitly control
+  the order.)
+
+## Invoke a method with interceptors
+
+Controller methods decorated with `@intercept` are invoked with applied
+interceptors for corresponding routes upon API requests.
```

Comment:
It's great to have a detailed documentation for interceptors :+1:

I feel the current structure is focused too much on low-level details and how things are implemented, rather on what a typical framework user needs to understand.

I am proposing to rework the page "Key concepts >> Interceptors" as follows:

1. Overview - keep it as it is
2. When are interceptors invoked - for controller methods 
3. How to apply interceptors in my code - use `@intercept` decorator
4. How to implement my own interceptor
5. Advanced - move the remaining information here

@strongloop/loopback-maintainers @bschrammIBM thoughts?